### PR TITLE
Feature/copy and move object

### LINKED ISSUE #57: Move/Copy object between buckets
**Copy object between buckets**

- [x]  objects less than 5GB

**Move object between buckets**

- [x]  objects less than 5GB

### PR DESCRIPTION
**Pull request by nettashviro**

### CODE REVIEW

File: object/handler.go [hunk 1]
```diff
@@ -219,3 +219,56 @@ func (h Handler) DeleteObjects(
...
+		ctx,
+		aws.String(request.GetBucketSrc()),
+		aws.StringSlice([]string{request.GetKeySrc()}),
+	)
+	if err != nil || len(deleteResponse.Errors) > 0 {
+		return nil, err
+	}
+
+	return &pb.MoveObjectResponse{Moved: request.GetKeySrc()}, nil
+}
```
Comment (Shahar-Y) [inline object/handler.go:274]:
```suggestion
}

```
Add newline at the end of the file

File: object/service.go [hunk 2]
```diff
@@ -431,3 +432,82 @@ func (s *Service) DeleteObjects(ctx aws.Context, bucket *string, keys []*string)
...
+
+	copyObjectResponse, err := s.s3Client.CopyObjectWithContext(ctx, copyObjectinput)
+	if err != nil {
+		return nil, fmt.Errorf("failed to copy object: %v", err)
+	}
+
+	// Compare the ETags between the source and the destination buckets (kind of a checksum)
+	if *sourceObjectResponse.ETag != *copyObjectResponse.CopyObjectResult.ETag {
+		return nil, fmt.Errorf(
+			"failed to copy object %s from source bucket, %s, because something went wrong in the process of copying the object to bucket %s and the ETag has changed : %v",
```
Comment (Shahar-Y) [inline object/service.go:504]:
```suggestion
			"failed to copy object %s from source bucket, %s, to destination bucket %s. ETag has changed : %v",
```
The error message is too long

File: object/object_test.go [hunk 3]
```diff
@@ -1710,6 +1711,561 @@ func TestHandler_DeleteObjects(t *testing.T) {
...
+			// reflect.DeepEqual is often incorrectly used to compare two like structs
+			// cmp.Equal is a better tool for comparing structs.
+			// https://golang.org/pkg/reflect/#DeepEqual
+			if got != nil && !cmp.Equal(tt.want, got, cmpopts.IgnoreUnexported(pb.MoveObjectResponse{})) {
+				t.Errorf("UploadHandler.MoveObject() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
```
Comment (Shahar-Y) [inline object/object_test.go:2268]: What about a test for dest bucket doesn't exist?
And key already exists in dest bucket?

**Review by Shahar-Y — CHANGES_REQUESTED** [review]
Reply to comments and don't resolve please.
All in all, great work :+1:

**Comment by Shahar-Y** [comment]
How does the copy handle changes to the wile while it is being copied (lock, etc...)?

File: object/handler.go (hunk 1 above)
Comment (nettashviro) [reply object/handler.go:274]: done

File: object/service.go (hunk 2 above)
Comment (nettashviro) [reply object/service.go:504]: done

File: object/object_test.go (hunk 3 above)
Comment (nettashviro) [reply object/object_test.go:2268]: f the destination bucket does not exist, a new bucket is created.

Comment (nettashviro) [reply object/object_test.go:2268]: The building of the object's key is in the file-service. The create/move function relies on the key sent by the file-service.

**Comment by nettashviro** [comment]
> How does the copy handle changes to the wile while it is being copied (lock, etc...)?
opened an issue #59

**Review by Shahar-Y — APPROVED** [review]

**Pull request merged**